### DISCUSSION TITLE
Accessing and Removing/Replacing/Deleting Zone Widgets

### DISCUSSION THREAD

**Discussion by smemis** [OrchardCMS/OrchardCore#13539] [General]
I want to clean/remove/delete the widgets added to the Footer Zone in Admin > Design > Widgets from the Contet__Page template and I want to add other widgets with WidgetListPart from this template. I cannot access Footer Zone from Content__Page template. How can I do that?

<pre>
Content__Page

{% assign xFooter = Model.ContentItem.Content.WidgetsListPart.Widgets.Footer %}

{% zone "Footer" %}

	{%if xFooter !=nil%}

		{% assign widgets = "Footer" | content_items:"Widget" %}
		{% for widget in widgets %}
			{% shape_remove_item "Footer" widget %}
		{% endfor %}

		{% for item in xFooter %}
			{{ item | shape_render }}
		{% endfor %}

	{%endif%}

{% endzone %}

{{ Model.Content | shape_render }}
</pre>

PS: I know the above codes are not correct but I wrote it to show what I am trying to do.

**Comment by ns8482e** [comment]
Not sure if I understand, zone widgets can also be placed at runtime using rules. If you need to add/remove widgets from a given zone, don't try to add/remove from the data instead, place them on zone using rules
